cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -2689,6 +2689,26 @@ TEST(TestArrowReadWrite, DictionaryColumnChunkedWrite) {
   ::arrow::AssertTablesEqual(*expected_table, *result, false);
 }
 
+TEST(TestArrowReadWrite, NonUniqueDictionaryValues) {
+  // ARROW-10237
+  auto dict_with_dupes = ArrayFromJSON(::arrow::utf8(), R"(["a", "a", "b"])");
+  auto indices = ArrayFromJSON(::arrow::int32(), R"([0, 1, 0, 1])");
```

Comment:
You might want to check all possible index values.

Reply:
Okay, I'll try all permutations of 0,1,2,null